### ISSUE TITLE
The "init" grains item always returns "sysvinit".

### ISSUE THREAD

**Issue by dumol** [saltstack/salt#37860]
### Description of Issue/Question

We have a setup with sysvinit / upstart / systemd Linux systems and we use the _init_ grain to deploy the appropriate init or unit file to these minions, change some system settings etc. However, with 2016.3.4, there's no way to differentiate among them. Below is the output for our 64-bit Linux minions (as the others have older Salt versions):
```
bs1f-lnx-centos5-x64-50:
    ----------
    init:
        sysvinit
    saltversion:
        2016.3.3
bs1i-lnx-debian7-x64-39:
    ----------
    init:
        sysvinit
    saltversion:
        2016.3.4
bs1a-lnx-ubuntu1404-x64-29:
    ----------
    init:
        sysvinit
    saltversion:
        2015.8.8.2
bs1a-lnx-arch-x64-45:
    ----------
    init:
        sysvinit
    saltversion:
        2016.3.4
bs1a-lnx-ubuntu1604-x64-46:
    ----------
    init:
        sysvinit
    saltversion:
        2016.3.2
bs1f-lnx-sles11-x64-51:
    ----------
    init:
        sysvinit
    saltversion:
        2015.8.10
bs1f-lnx-centos7-x64-48:
    ----------
    init:
        sysvinit
    saltversion:
        2016.3.3
bs1i-lnx-sles12-x64-42:
    ----------
    init:
        sysvinit
    saltversion:
        2015.8.7
bs1f-lnx-centos6-x64-49:
    ----------
    init:
        sysvinit
    saltversion:
        2016.3.3
```

### Setup

A relevant state affected by this issue:
```
#
# This disables the graphical login for Linux minions.
# Covers RHEL 5-7, SLES 11-12, Ubuntu 12.04/14.04, Debian 7 and Arch.
#

{% set config_file = "/etc/inittab" %}
{% set old_pattern = "^id:5:initdefault" %}
{% if grains["os"] == "Debian" %}
    {% set new_pattern = "id:2:initdefault" %}
{% else %}
    {% set new_pattern = "id:3:initdefault" %}
{% endif %}

# In RHEL 6 (using Upstart), the init level is still set trough /etc/inittab.
{% if grains["init"] == "upstart" and not grains ["oscodename"] == "Santiago" %}
    {% set config_file = "/etc/init/rc-sysinit.conf" %}
    {% set old_pattern = "^env DEFAULT_RUNLEVEL=5$" %}
    {% set new_pattern = "env DEFAULT_RUNLEVEL=2" %}
{% endif %}

Do not boot in graphical mode:
{% if grains["init"] == "systemd" %}
  file.symlink:
    - name: /etc/systemd/system/default.target
    - target: /lib/systemd/system/multi-user.target
{% else %}
  file.replace:
    - name: {{config_file}}
    - pattern: {{old_pattern}}
    - repl: {{new_pattern}}
    - backup: False
{% endif %}
```

### Steps to Reproduce Issue
Not sure if relevant, this is the debug log from the CentOS 7 minion (using systemd):
```
2016-11-23 14:22:05,256 [salt.minion      ][INFO    ][2164] User salt Executing command grains.item with jid 20161123142205244832
2016-11-23 14:22:05,258 [salt.minion      ][DEBUG   ][2164] Command details {'tgt_type': 'glob', 'jid': '20161123142205244832', 'tgt': 'bs*', 'ret': '', 'user': 'salt', 'arg': ['init'], 'fun': 'grains.item'}
2016-11-23 14:22:05,286 [salt.minion      ][INFO    ][10875] Starting a new job with PID 10875
2016-11-23 14:22:05,289 [salt.utils.lazy  ][DEBUG   ][10875] LazyLoaded grains.item
2016-11-23 14:22:05,291 [salt.utils.lazy  ][DEBUG   ][10875] LazyLoaded direct_call.get
2016-11-23 14:22:05,291 [salt.minion      ][DEBUG   ][10875] Minion return retry timer set to 10 seconds (randomized)
2016-11-23 14:22:05,292 [salt.minion      ][INFO    ][10875] Returning information for job: 20161123142205244832
2016-11-23 14:22:05,293 [salt.transport.zeromq][DEBUG   ][10875] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'bs1f-lnx-centos7-x64-48', 'tcp://172.20.0.24:4506', 'aes')
2016-11-23 14:22:05,293 [salt.crypt       ][DEBUG   ][10875] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'bs1f-lnx-centos7-x64-48', 'tcp://172.20.0.24:4506')
```

### Versions Report

Master version (the minions are not that relevant, I suppose, but you may see their versions above, in the first output pasted):
```
Salt Version:
           Salt: 2016.3.4
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-101-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```
Thanks!

**Comment by Ch3LL** [comment]
@dumol i'm hvaing  a hard time replicating this on a centos7 minion as shown below:

```
[root@localhost salt]# salt-call --local grains.item init
local:
    ----------
    init:
        systemd
[root@localhost salt]# salt --version
salt 2016.3.4 (Boron)
```

as shown above it is correctly displaying systemd. What version was this working on previously? and any other information to possibly help replicate this?

**Comment by dumol** [comment]
By the way, the issue is still present on our minion after upgrading to the latest available version: 2016.3.4. 

Unfortunately, I can't tell when it was last working for us, I think it was a matter of weeks. And it doesn't seem to be related to the running versions, I now realize we have minions running 2015.5.3 that exhibit the problem and minions running 2016.3.4 that are fine.  ¯\_(ツ)_/¯

**Comment by dumol** [comment]
Hold on, I think I've identified a malfunction in our Salt infrastructure that would explain this. It seems I had to set the `init` grain for a SLES 11 minion, but that state was applied to every minion when applying `highstate` because of an error in the `top.sls` file. 

Closing this… And sorry for the bother!
